Changed the project and added necessary protocols

BJap opened the pull request:
Switched to use a framework not a static library for Swift support
Added protocols for entire framework

Commit:
Switched to use a framework not a static library for Swift support
Added protocols for entire framework

Commit:
Switching to Allman Style from 1TBS

BJap commented on SwiftCrypt/SwiftCrypt/Protocols/Cipher.swift:
```diff
@@ -0,0 +1,39 @@
+{    
+    /// Encrypts the text.
+    ///
+    /// - Parameter clearText: The text to be encrypted.
```
Copy and paste error

BJap commented on SwiftCrypt/SwiftCrypt/Protocols/Key.swift:
```diff
@@ -0,0 +1,33 @@
+{    
+    /// Generates a random key.
+    ///
+    /// - Parameter clearText: The text to be prepared.
```
Copy and paste error

SD10 commented:
An improvement could be to comply with Swift 3 API design guidelines. Specifically regarding parameter naming. For example:

`static func decrypt(cipherText: String, key: String) -> String`

Could be renamed to:

`static func decrypt(cipherText: String, withKey key: String) -> String`

This reduces ambiguity at the call site:

`decrypt(cipherText: String, withKey: String)`

Opposed to:

`decrypt(cipherText: String, key: String)`

Suggested changes:

// Line 25 of Key protocol
`static func randomKey(withSeed seed: UInt32) -> String`

// Line 20 of Cipher protocol
`static func encrypt(clearText: String, withKey key: String) -> String`

// Line 29 of Cipher protocol
`static func decrypt(cipherText: String, withKey key: String) -> String`

For more information on Swift 3 API Design Guidelines refer to:
https://swift.org/documentation/api-design-guidelines/#naming

Commit:
Moved the project up to root level
Made changes to function wording according to Swift 3 best practices
Removed a copy paste addition extra in one function documentation
